opm/simulators/flow/partitionCells.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+  Copyright 2021 Total SE
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/simulators/flow/partitionCells.hpp>
+
+#include <fmt/format.h>
+
+#include <algorithm>
+#include <cassert>
+#include <fstream>
+#include <iterator>
+#include <numeric>
+#include <string>
+
+namespace Opm
+{
+
+
+    // Read from file, containing one number per cell.
+    std::pair<std::vector<int>, int> partitionCellsFromFile(const std::string& filename, const int num_cells)
+    {
+        // Read file into single vector.
+        std::ifstream is(filename);
+        const std::vector<int> cellpart{std::istream_iterator<int>(is), std::istream_iterator<int>()};
+        if (cellpart.size() != size_t(num_cells)) {
+            auto msg = fmt::format("Partition file contains {} entries, but there are {} cells.",
+                                   cellpart.size(), num_cells);
+            throw std::runtime_error(msg);
```

Comment:
You should probably bring in `<stdexcept>` for a declaration of the `std::runtime_error` class.